src/BSON/Binary.c

```diff
@@ -352,9 +352,8 @@ static int php_phongo_binary_compare_objects(zval* o1, zval* o2) /* {{{ */
 static HashTable* php_phongo_binary_get_gc(zval* object, zval** table, int* n) /* {{{ */
```

Comment:
Given that all `*_get_gc` functions do exactly the same, would it make sense to create a generic `php_phongo_get_gc` function and use that for all classes that need this functionality?

Reply:
Good idea. I've amended the patch, which reduced quite a bit of copypasta.

This change now has us define the common `get_gc` handler for all of our objects, which I don't expect will be a problem. It's also more future proof in the event that other classes define a `get_properties` handler down the line.